broker-daemon/utils/logger.js

```diff
@@ -13,9 +13,41 @@
 
 const winston = require('winston')
 
+/**
+ * List of properties whose contents should be filtered from logs.
+ * @constant
+ * @type {Array}
+ */
+const SENSITIVE_PROP_LIST = Object.freeze([
+  'username',
+  'pass',
+  'password',
+  'passphrase',
+  'recoverySeed'
+])
+
+/**
+ * String to replace sensitive data with.
+ * @constant
+ * @type {String}
+ */
+const SENSITIVE_REPLACEMENT = '***FILTERED***'
+
+/**
+ * Formatter to filter sensitive data from logs.
+ */
+const filterSensitive = winston.format((info) => {
+  return SENSITIVE_PROP_LIST.reduce((info, key) => {
```

Comment:
Okay this might be annoying, but what initialize and empty object and if you iterate through the info object instead of the SENSITIVE_PROP_LIST.

That way there is no mutation and you perform the same behavior on all the key values, and just filter when the key is in the SENSITIVE_PROP_LIST?